### DISCUSSION TITLE
How to automatically hide build files (.props and .targets) and build folder from VS solution explorer.

### DISCUSSION THREAD

**Discussion by elTRexx** [NuGet/Home#12861] [Q&A] [answered]
Hi.
I'm very new to nuget package creation.
I have created my first one as an incremental source generator. So my package only appear in Analyzer in solution explorer inside Visual Studio and it is fine.
But now I want to add a file inside my package to be copied on the consuming project.
I managed to address this by using a  custom .props file inside a build folder of my package.
It work as I wanted, but now, the consuming project solution explorer view show my package and inside it build files folder along my .props file.
Consuming project could add `ExcludeAssets="Compile"` in order to hide them, but it is a manual task. How can it be automated from my package creator point of view so my consuming project won't have to bother hiding those build files in their solution explorer view.
 
Note : I saw this [issue](https://github.com/NuGet/Home/issues/4856 ) witch look similar to mine (this is about contentFiles), but solution didn't work for my build files.

![](https://i.stack.imgur.com/43BvD.png)

Here is a [link](https://stackoverflow.com/questions/76989354/hide-embedded-nuget-files-from-vs-solution-explorer) to the same question i asked on StackOverflow.

**Comment by zivkan** [comment, marked as answer]
The only way to avoid it showing up in Solution Explorer, is to remove it from your package.

It's shown in order to provide develoeprs insight into how each package is affecting their project. It allows them to inspect the package's files and validate it's not doing anything unwanted or malicious. The package readme, in particular, [is supposed to educate package consumers](https://devblogs.microsoft.com/nuget/write-a-high-quality-readme-for-nuget-packages/) about what the package does, how to use it, and where to get more information, so it doesn't make sense to me why you'd include a readme, but then want to hide it.

If `ExcludeAssets="compile"` is preventing the package's readme and/or build assets from being shown, that's a bug we have to fix. `ExcludeAssets="compile"` should only present dlls in the package's `lib/` or `ref/` directories from being used by the project. The package's readme should always be displayed, the the MSBuild props file is selected by a different asset group, not `compile`.

**Comment by elTRexx** [comment]
I agree for Readme, it is fine by me to be exposed here since it is meant to be read. 
But the .props is a inner, non modifiable, building mechanism that I think majority of consumer don't care about. It is still present to do what it is meant to do, but showing it kind of "pollute" the explorer view, no ?
Though I completely endorse your concern about malicious building actions.

I mean I'd rather have to be hidden by default, since, in my opinion, the majority of consumer wouldn't care about it, but allowing it to be visible for, and by, any consumer that want to see it.
